src/ApiHandler.ts

```diff
@@ -139,6 +125,69 @@ export default class ApiHandler {
 			}
 		}
 
+		const byteFee = api.consts.transactionPayment.transactionByteFee;
+		const extrinsicBaseWeight = api.consts.system.extrinsicBaseWeight;
+		const coefficients = api.consts.transactionPayment.weightToFee[0];
+		const coeffFrac = coefficients.coeff_frac;
+		const coeffInt = coefficients.coeff_integer;
+		
+		// TODO: generalize for terms that may have more degrees & may be negative
+		const degree = coefficients.degree;
+		const negative = coefficients.negative;
+
+		for (let idx = 0; idx < block.extrinsics.length; ++idx) {
+			const extrinsic = block.extrinsics[idx];
+			if (!extrinsic.isSigned) {
+				continue;
+			}
+
+			try {
+				if (api.runtimeVersion.specName.toString() === 'kusama') {
+					extrinsics[idx].info = await api.rpc.payment.queryInfo(extrinsic.toHex(), parentHash);
+				} else {
+					const parsedXt = extrinsics[idx];
+					const xtEvents = parsedXt.events;
+					const success = 'system.ExtrinsicSuccess';
+					const failure = 'system.ExtrinsicFailed';
+					const completedEvent = xtEvents.find((event) => event.method === success || event.method === failure);
+					if (!completedEvent) {
+						parsedXt.info = { error: 'Unable to fetch fee info' };
+						continue;
+					}
+
+					const completedData = completedEvent.data;
+					if (!completedData) {
+						parsedXt.info = { error: 'Unable to fetch fee info' };
+						continue;
+					}
+
+					const weightInfo = completedData[completedData.length - 1] as DispatchInfo;
+					if (!weightInfo.weight) {
+						parsedXt.info = { error: 'Unable to fetch fee info' };
+						continue;
+					}
+
+					const weight = weightInfo.weight;
+					const encodedLength = extrinsic.encodedLength;
+					const multiplier = await api.query.transactionPayment.nextFeeMultiplier.at(parentHash);
+
+					const multiplied = multiplier.mul(weight.mul(coeffFrac)
+                                       .add(weight.mul(coeffInt))
+                                       .add(byteFee).muln(encodedLength));
```

Comment:
That is barely readable. Parentheses that start on the same line should close on the same line. I also think that it is wrong (the whole sum was multiplicated with  `encodedLength`)
```suggestion
					const multiplied = multiplier.mul(
					    weight.mul(coeffFrac)
                                               .add(weight.mul(coeffInt))
                                               .add(byteFee.muln(encodedLength))
                                        );
```

Reply:
The only thing that is `Perbill` is `coeffFrac`. The rest are integers.

Reply:
+1 for readable code...sorry about that. I do not claim to be a math wizard, so please let me know if I have accurately implemented your calculations :pray: